pkg/cmdutil/repo_override.go

```diff
@@ -51,20 +51,49 @@ func EnableRepoOverride(cmd *cobra.Command, f *Factory) {
 		if err := executeParentHooks(cmd, args); err != nil {
 			return err
 		}
-		repoOverride, _ := cmd.Flags().GetString("repo")
-		f.BaseRepo = OverrideBaseRepoFunc(f, repoOverride)
+
+		// First, get the value of the flag.
+		// We can ignore errors here because the flag is guaranteed to exist and be string.
+		userProvidedRepo, _ := cmd.Flags().GetString("repo")
+		if userProvidedRepo == "" {
+			// If there was no flag set, then check the GH_REPO environment variable,
+			// which has lower precedence.
+			repoOverride, ok := os.LookupEnv("GH_REPO")
+			// If the envrionment variable was set, then set the value of the `repo` flag,
+			// this ensures that checks for `HasChanged` work correctly. It's a bit "spooky"
+			// action at a distance because the flag will not have been set by the user,
+			// but perhaps it makes sense?
+			//
+			// The reason we need this is because there are `secret` commands that need to know whether
+			// the user set the repo, and the alternative is them having knowledge of the flag and env var,
+			// or by adjusting all our types so that the "source" of the BaseRepo is surfaced, which is a
+			// pretty big change.
+			if ok {
```

Comment:
Nothing wrong with the logic; just stating how this works.

If a user has `GH_REPO=""`, then we will attempt to set the flag with the value if empty and [defer to `spf13/pflag` to determine if the flag has been changed or not](https://github.com/spf13/pflag/blob/5ca813443bd2a4d9f46a253ea0407d23b3790713/flag.go#L462-L495).